src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/ShortTermCustodyPredictorService.kt

```diff
@@ -5,33 +5,84 @@ import hex.genmodel.easy.RowData
+                "${predictorParameters.offenderAge}, " +
+                "${predictorParameters.offenceCode})")
 
         val rowData = RowData()
 
-        rowData["court_name"] = courtName
-        offenderAge?.let {
-            rowData["offender_age"] = offenderAge.toString()
+        rowData["court_name"] = predictorParameters.courtName
+        predictorParameters.offenderAge?.let {
+            rowData["offender_age"] = predictorParameters.offenderAge.toString()
         }
-        rowData["offence_code"] = offenceCode
+        rowData["offence_code"] = predictorParameters.offenceCode
 
         val prediction = model.predictBinomial(rowData)
         val score = prediction.classProbabilities[0]
-        log.info("Calculated short term custody score of $score for court: $courtName, offender age: $offenderAge, offence code: $offenceCode")
+        log.info("Calculated short term custody score of $score for court: ${predictorParameters.courtName}, offender age: ${predictorParameters.offenderAge}, offence code: ${predictorParameters.offenceCode}")
 
         return score
     }
+
+    fun addPredictorScoresToHearing(hearingEntity: HearingEntity) {
+        log.debug("Entered addPredictorScoresToHearing for hearing with case number: ${hearingEntity.caseNo}")
+
+        hearingEntity.hearingDefendants.forEach { defendant ->
+            defendant.offences.forEach { offence ->
+                val homeOfficeOffenceCode : String? =
+                try {
+                    offence.offenceCode?.let { offencesRestClient.getHomeOfficeOffenceCodeByCJSCode(it) }
+                } catch (e : ExternalApiEntityNotFoundException) {
+                    log.warn("No corresponding Home office offence code could be found for CJS code: ${offence.offenceCode}", e)
+                    null
+                } catch (e : ExternalApiUnknownException) {
+                    log.error("Unknown error occurred whilst looking up Home office offence code for CJS code: ${offence.offenceCode}", e)
+                    null
+                }
+
+                if (homeOfficeOffenceCode != null) {
+                    log.debug("Found Home office offence code: $homeOfficeOffenceCode")
+                    courtRepository.findByCourtCode(hearingEntity.hearingDays[0].courtCode).ifPresentOrElse({
+                        val score = calculateShortTermCustodyPredictorScore(
+                            PredictorParameters(
+                                it.name,
+                                calculateAge(defendant),
+                                homeOfficeOffenceCode
+                            )
+                        )
+                        offence.shortTermCustodyPredictorScore = BigDecimal.valueOf(score)
+                    }, {
+                        log.warn("No court name could be found for court code: ${hearingEntity.hearingDays[0].courtCode}")
```

Comment:
Nothing wrong here but could this be moved into a separate function instead? From try/catch block onwards.